compactor/src/query.rs

```diff
@@ -158,8 +158,11 @@ impl QueryChunk for QueryableParquetChunk {
     // it is not safe to use it for sorting chunk
     fn id(&self) -> ChunkId {
         let timestamp_nano = self.min_time.get();
-        let timestamp_nano_u128 =
-            u128::try_from(timestamp_nano).expect("Cannot convert timestamp nano to u128 ");
+        let error_msg = format!(
+            "Cannot convert timestamp nano to u128. Timestamp nano: {}, Paritition id: {}",
+            timestamp_nano, self.partition_id
+        );
+        let timestamp_nano_u128 = u128::try_from(timestamp_nano).expect(&error_msg);
```

Comment:
This will always allocate & generate the string message, even when it's not needed.

If we do this instead:

```suggestion
        let timestamp_nano_u128 = u128::try_from(timestamp_nano).unwrap_or_else(|_| panic!(
            "Cannot convert timestamp nano to u128. Timestamp nano: {}, Paritition id: {}",
            timestamp_nano, self.partition_id
        ));
```

The string will only be generated when it is needed :+1:

Reply:
We could also include the error, if that is useful:

```rust
let timestamp_nano_u128 = u128::try_from(timestamp_nano).unwrap_or_else(|e| panic!(
    "Cannot convert timestamp nano to u128. Timestamp nano: {}, Paritition id: {}, error: {}", timestamp_nano, self.partition_id, e
));
```